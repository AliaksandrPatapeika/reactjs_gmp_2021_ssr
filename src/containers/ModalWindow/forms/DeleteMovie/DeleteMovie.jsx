import PropTypes from 'prop-types';
import React from 'react';
import {useDispatch, useSelector} from 'react-redux';

import {deleteMovie} from '../../../../actions/movies';
import Button from '../../../../components/Button';

const DeleteMovie = ({formTitle, movieId}) => {
  const activeModalWindow = useSelector((state) => state.movie.activeModalWindow);
  const dispatch = useDispatch();
  const onSubmit = (event) => {
    event.preventDefault();
    dispatch(deleteMovie(movieId));
  };

  return activeModalWindow === 'deleteMovie' && (
  // PATTERN: Use Fragments to Avoid Additional HTML Element Wrappers
  <>
    <span className="title">{formTitle}</span>
    <form onSubmit={onSubmit}>
      <h1 className="content">Are you sure you want to delete this movie?</h1>
      <div className="modalFooter">
        <Button
          type="submit"
          className="button"
          title="CONFIRM"
        />
      </div>
    </form>
  </>
  );
};

DeleteMovie.propTypes = {
  formTitle: PropTypes.string.isRequired,
  movieId: PropTypes.number.isRequired
};

export default DeleteMovie;
