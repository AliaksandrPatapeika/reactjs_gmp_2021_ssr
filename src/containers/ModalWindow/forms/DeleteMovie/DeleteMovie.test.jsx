import {head} from 'lodash';
import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import MockMovies from '../../../../tests/mocks/mockData';
import DeleteMovie from './DeleteMovie';

describe('DeleteMovie', () => {
  test('renders correctly', () => {
    const mockMovie = head(MockMovies);
    const mockStore = configureMockStore();
    const initialState = {
      movie: {
        activeModalWindow: 'deleteMovie'
      }
    };
    const store = mockStore(initialState);
    const deleteMovie = renderer
      .create(
        <Provider store={store}>
          <DeleteMovie formTitle="title" movieId={mockMovie.id} />
        </Provider>
      )
      .toJSON();

    expect(deleteMovie).toMatchSnapshot();
  });
});
