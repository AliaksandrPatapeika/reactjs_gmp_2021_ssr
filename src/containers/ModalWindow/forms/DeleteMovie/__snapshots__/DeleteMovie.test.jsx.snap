// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMovie renders correctly 1`] = `
Array [
  <span
    className="title"
  >
    title
  </span>,
  <form
    onSubmit={[Function]}
  >
    <h1
      className="content"
    >
      Are you sure you want to delete this movie?
    </h1>
    <div
      className="modalFooter"
    >
      <button
        className="button"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="submit"
      >
        CONFIRM
        
      </button>
    </div>
  </form>,
]
`;
